plugins/pci-bcr/fu-plugin-pci-bcr.c

```diff
@@ -11,20 +11,45 @@
 
 struct FuPluginData {
 	gboolean		 has_device;
+	guint8			 bcr_addr;
 	guint8			 bcr;
 };
 
-#define BCR			0xdc
 #define BCR_WPD			(1 << 0)
 #define BCR_BLE			(1 << 1)
 #define BCR_SMM_BWP		(1 << 5)
 
 void
 fu_plugin_init (FuPlugin *plugin)
 {
-	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+	FuPluginData *priv = fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
 	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
 	fu_plugin_add_udev_subsystem (plugin, "pci");
+
+	/* this is true except for some Atoms */
+	priv->bcr_addr = 0xdc;
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+	guint32 eax = 0;
+	guint64 tmp;
+	g_autofree gchar *group = NULL;
+
+	/* has a quirk */
+	if (!fu_common_cpuid (0x1, &eax, NULL, NULL, NULL, error))
+		return FALSE;
+	group = g_strdup_printf ("DeviceInstanceId=CPUID\\PRO_%01X&FAM_%01X&MOD_%01X",
```

Comment:
As a another thought - shouldn't this populate from the cpu plugin?  Like CPU plugin can carry the quirk and some metadata w/ `BcrAddr`  And this plugin can lookup the device from the CPU plugin?

Reply:
So we'd add the GUIDs to the `FuCpuDevice` and then set up a `fu_plugin_device_registered()` in the `pci-bcr` plugin, then check if the device has some kind of metadata using `fu_device_get_metadata_integer()` and if so set the `bcr_addr`?

Reply:
Exactly!

Reply:
I've updated the patch with this suggestion; I agree it's a better design.